modules/internal/StringCasts.chpl

```diff
@@ -259,8 +275,10 @@ module StringCasts {
 
 
   inline proc _cast(type t:chpl_anycomplex, x: string) throws {
+    pragma "fn synchronization free"
     pragma "insert line file info"
     extern proc c_string_to_complex64(x:c_string, ref err: bool) : complex(64);
+    pragma "fn synchronization free"
```

Comment:
Can we split adding `pragma "fn synchronization free"` to all these functions into a separate PR?